### PR TITLE
feat(build): Fixing user home bug

### PR DESCRIPTION
**Pull request by blee1170**
There is a issue when running on alpine linux. It will error when trying to find the users home directory:
```
Could not read current user from environment, failing.
user: Current not implemented on linux/amd64
```
This PR will first try to use user.Current, if that fails then it will attempt to get the home directory from the environment.

### CODE REVIEW

File: command/meta.go [hunk 1]
```diff
@@ -112,21 +112,30 @@ func (m *ApiMeta) Process(args []string) ([]string, error) {
...
-	m.configLocation = filepath.Join(usr.HomeDir, ".spin", "config")
+	m.configLocation = filepath.Join(userHome, ".spin", "config")
 	yamlFile, err := ioutil.ReadFile(m.configLocation)
 	if err != nil {
 		m.Ui.Warn(fmt.Sprintf("Could not read configuration file from %s.", m.configLocation))
 	}
 
 	if yamlFile != nil {
-		err = yaml.UnmarshalStrict(yamlFile, &m.Config)
+		err = yaml.Unmarshal(yamlFile, &m.Config)
```
Comment (jtk54) [inline command/meta.go]: Is there a reason this changed?

Comment (blee1170) [reply command/meta.go]: For me it was error'ing when I did `go build`. I didn't really intent for that to be part of this PR. I can switch it back if need be.

Comment (jtk54) [reply command/meta.go]: If the parsing was failing, either the config is malformed or spin isn't tolerant enough. In either case it seems unrelated to this PR.

Comment (blee1170) [reply command/meta.go]: Okay. I will update the PR.

**Comment by jtk54** [comment]
Thanks for the contribution :+1:

**Pull request merged**